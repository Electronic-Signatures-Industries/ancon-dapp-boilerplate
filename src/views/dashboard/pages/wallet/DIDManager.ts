
import { Ed25519Provider } from "key-did-provider-ed25519";
import KeyResolver from "@ceramicnetwork/key-did-resolver";
import { DID } from "dids";
import { arrayify, mnemonicToSeed } from "ethers/lib/utils";

export class DIDManager {
  /**
   * Create 3ID
   * using XDV
   * @param wallet
   * @param messageNotify
   */
  async create3ID(wallet) {
    let seed = arrayify(mnemonicToSeed(wallet.mnemonic));
    seed = seed.slice(0, 32);
    const provider = new Ed25519Provider(seed);
    const did = new DID({ provider, resolver: KeyResolver.getResolver() });
    await did.authenticate();
    return did;
  }
}
