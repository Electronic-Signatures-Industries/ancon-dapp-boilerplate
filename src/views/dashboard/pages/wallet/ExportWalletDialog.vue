<template>
  <v-dialog v-model="show" max-width="500px">
    <v-card>
      <v-card-title>
        <span class="headline">Upload and sign documents</span>
      </v-card-title>

      <v-card-text>
        <v-form v-model="form" autocomplete="off">
          <v-row>
            <v-col cols="12" md="12">
              <v-file-input
                prepend-icon="mdi-paperclip"
                v-model="value"
                multiple
                show-size
                label="Files"
              ></v-file-input>
            </v-col>
          </v-row>
        </v-form>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click="show = false">Cancel</v-btn>
        <v-btn color="blue darken-1" text @click="change">OK</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>
<script lang="ts">

import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  name: 'xdv-upload',
  props: ['value', 'show'],
})
export default class Upload extends Vue {
  value: string = '';
  show = false;
   change() {
    this.$emit('input', this.value);
  }
}
</script>
