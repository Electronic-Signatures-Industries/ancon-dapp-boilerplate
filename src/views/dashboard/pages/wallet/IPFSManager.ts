import IPFS from 'ipfs-core'
import dagJose from 'dag-jose'
import multiformats from 'multiformats/cjs/src/basics'
import legacy from 'multiformats/cjs/src/legacy'
import { DID } from 'dids'
import { ethers } from 'ethers'
import moment from 'moment'
import axios from 'axios'
import IPFSClient from 'ipfs-http-client';
import { SwarmNodeSignedContent } from '../shared/SwarmNodeSignedContent'
import { keccak256 } from 'ethers/lib/utils'
const dagCBOR = require('ipld-dag-cbor');
const Ipld = require('ipld')
const IpfsRepo = require('ipfs-repo')
const IpfsBlockService = require('ipfs-block-service')
const multicodec = require('multicodec')
const Repo = require('ipfs-repo');

export type DocumentMetadata = any;

const initIpld = async (repo) => {
  //const repo = new IpfsRepo(ipfsRepoPath)
  //await repo.init({})
  //await repo.open()
  const blockService = new IpfsBlockService(repo)
  return new Ipld({blockService: blockService})
}

const MAINNET = `https://mainnet.infura.io/v3/92ed13edfad140409ac24457a9c4e22d`;
export class IPFSManager {
    format = 'dag-cbor';
    client: IPFS.IPFSRepo;
    provider: ethers.providers.JsonRpcProvider;
    ipld: any;

    async start(){
       this.client = IPFSClient({ url: `https://ipfs.xdv.digital` });
       this.ipld = await initIpld(this.client);
    }

    async stop() {
        //await this.client.stop();
    }

    /**
     * Converts Blob to Keccak 256 hash
     * @param payload 
     */
    async blobToKeccak256(payload: Blob): Promise<string> {
        let ab = await payload.arrayBuffer();
        let buf = new Uint8Array(ab);
        return keccak256(buf) as string;
    }

    async setCurrentNode(
        cid: string,
        key: string = 'self',

    ) {
        const res = await this.client.name.publish(cid, {key});
        return res;
    }

    async getCurrentNode(
        key: string = 'self'
    ) {
        try {
            let query = '';
            const keys = await this.client.key.list();
            const { id } = keys.find(i => i.name === key);
            for await (query of this.client.name.resolve(`/ipns/${id}`)) {
            }
            return query.replace('/ipfs/', '');
        } catch (e) {
            return null;
        }
    }

    /**
     * Add Signed Object
     * @param did DID
     * @param payload Payload, either Buffer or Blob 
     * @param previousNode If it has previous node
     */
    async addSignedObject(
        did: DID,
        payload: File,
        previousNode?: any) {
        let temp: string;
        let content: Buffer;
        if (payload instanceof File) {
            temp = await this.blobToKeccak256(payload);
            content = Buffer.from((await payload.arrayBuffer()));
        } else {
            throw new Error('addSignedObject: must be a file object');

        }
        temp = temp.replace('0x', '');
        // sign the payload as dag-cbor
        //* display on grid
        const { jws, linkedBlock } = await did.createDagJWS({
            content: content.toString('base64'),//*
        });
        const metadata = {
            contentType: payload.type,
            name: payload.name, //*
            lastModified: payload.lastModified,//*
            timestamp: moment().unix(),
            hash: temp,//*
            id: atob(moment().unix() + temp),
            documentPubCert: undefined,
            documentSignature: undefined,
            signaturePreset: undefined,
            contentRef: undefined
        }
        // put the JWS into the ipfs dag
        const jwsCid = await this.client.dag.put(jws, multicodec.DAG_CBOR);
        // put the payload into the ipfs dag
        await this.client.block.put(linkedBlock, { cid: jws.link })
        console.log('cid', jwsCid.toString());
        metadata.contentRef = jwsCid.toString();
        return metadata;
    }

    createSignedContent({
        contentType,
        name,
        lastModified,
        size,
        content,
        hash,
        documentPubCert,
        documentSignature,
        signaturePreset
    }) {
        return {
            contentType,
            name,
            lastModified,
            size,
            content,
            hash,
            created: moment().unix(),
            documentPubCert,
            documentSignature,
            signaturePreset,
        } as SwarmNodeSignedContent;

    }

    async addIndex(
        did: DID,
        documents: any[]) {
        // sign the payload as dag-cbor
        const { jws, linkedBlock } = await did.createDagJWS({
            documents
        });
        
        // put the JWS into the ipfs dag
        const jwsCid = await this.client.dag.put(jws, multicodec.DAG_CBOR);
        // put the payload into the ipfs dag
        
        await this.client.block.put(linkedBlock, { cid: jws.link });
        const cid = jwsCid.toString()
        return cid;
    }
    
    /**
     * Get IPLD object
     * @param cid content id
     */
    async getObject(cid: string): Promise<any> {
        let temp = await this.client.dag.get(cid);
        const res = {
            metadata: {
                ...temp,
            },
            payload: undefined
        }
        temp = await this.client.dag.get(cid, { path: '/link' });
        res.payload = {
            ...temp,
        };

        return temp;
    }

    verify(did: DID, obj: any): Promise<any> {
        return did.verifyJWS(obj.metadata);
    }

    async encryptObject(did: DID, cleartext, dids: string[]) {
        const jwe = await did.createDagJWE(cleartext, dids)
        return this.client.dag.put(jwe, multicodec.DAG_CBOR);
    }

    async decryptObject(did: DID, cid, query) {
        const jwe = (await this.client.dag.get(cid, query)).value
        const cleartext = await did.decryptDagJWE(jwe)
        return cleartext;
    }
    async addPublicWallet(
        did: DID,
        payload: Buffer) {
        let temp: string;
        let content: Buffer;
        temp = keccak256(payload);
        content = payload;
        temp = temp.replace('0x', '');
        // sign the payload as dag-cbor
        const cid = await this.client.add({
            path: 'index.json',
            content
        });
        return cid.toString()
    }

    /**
     * Creates key
     * @param name name 
     * @param password password
     */
    async createKey(name: string, password: string) {
        const key = await this.client.key.gen(name, {
            type: 'rsa',
            size: 2048
        });

        const pem = await this.client.key.export(name, password);

        return { name, pem };
    }

}