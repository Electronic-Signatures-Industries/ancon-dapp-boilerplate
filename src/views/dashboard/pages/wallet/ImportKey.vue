<template>
  <v-dialog v-model="show" max-width="500px">
    <v-card>
      <v-card-title>
        <span class="headline">Passphrase</span>
      </v-card-title>

      <v-card-text>
        <v-form v-model="form" autocomplete="off">
          <v-row>
            <v-col cols="12" md="12">
              <v-text-field
                required
                v-model="value"
                :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
                :type="showPassword ? 'text' : 'password'"
                class="input-group--focused"
                @click:append="showPassword = !showPassword"
                :error="validations.value"
              ></v-text-field>
            </v-col>
          </v-row>
        </v-form>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click="show = false">Cancel</v-btn>
        <v-btn color="blue darken-1" text @click="change">OK</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>
<script lang="ts">
import { TypedRFE, TasaISC, ISC } from 'ifesa-dgi-factura-electronica';
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  name: 'xdv-unlock',
  props: ['value', 'show'],
})
export default class Unlock extends Vue {
  value: string = '';
  show = false;
  showPassword = false;


  validations: any = { password: false };
  change() {
    this.$emit('input', this.value);
  }
}
</script>
