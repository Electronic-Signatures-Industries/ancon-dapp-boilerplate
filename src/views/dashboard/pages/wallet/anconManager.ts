import { base64, keccak256 } from 'ethers/lib/utils'
import { AnconClient } from 'anconjs'
import { config, directive } from 'vue/types/umd'
import {
  MsgFileResponse,
  MsgMetadataResponse,
  MsgMetadataTx,
} from 'anconjs/generated/Electronic-Signatures-Industries/ancon-protocol/ElectronicSignaturesIndustries.anconprotocol.anconprotocol/module/types/anconprotocol/tx'
import { firstValueFrom, Observable, Subject, Subscription } from 'rxjs'
import * as cosmosConfig from './anconConfig'

export type DocumentMetadata = any

export class AnconManager {
  anconClient: AnconClient
  anconAPI: any
  account: any
  did: any

  //xdvWallet: Wallet;

  async start(passphrase: string) {
    this.anconClient = new AnconClient(
      true,
      cosmosConfig.default.rest,
      cosmosConfig.default.rpc,
    )

    this.anconAPI = await this.anconClient.create(
      'ancon_manager_acct',
      passphrase,
      passphrase,
    )

    const wallet = this.anconClient.wallet
    const acct = (await wallet.getAccount()) as any
    const eddsa = await wallet.createEd25519({
      accountName: 'ancon_manager_accct',
      passphrase,
      walletId: acct.keystores[0].walletId,
    })
    this.did = eddsa.did
    this.account = this.anconClient.account[0].address
  }
  /**
   * Converts Blob to Keccak 256 hashAA
   * @param payload
   */
  async blobToKeccak256(payload: File): Promise<any> {
    let ab = await payload.arrayBuffer()
    let buf = new Uint8Array(ab)
    let hash = keccak256(buf)
    let content = Buffer.from(await payload.arrayBuffer())

    return { hash, content }
  }

  /**
   * Add Ancon Metadata
   * @param name Identifies the asset to which this token represents.
   * @param description Describes the asset to which this token represents.
   * @param image A URI pointing to a resource with mime type image/* representing the asset to which this token represents.
   * @param sources Current intellectual property.
   * @param owner The owner is a DID identifier.
   * @param parent Direct ascendant of the current intellectual property.
   * @param verifiedCredentialRef Is the verified credential for the metadata
   * @param links Sample of references included in the current intellectual property
   * @param creator
   * @param did
   * @param from
   */
  async addMetadata({
    name,
    description,
    image,
    sources,
    owner,
    parent,
    verifiedCredentialRef,
    links,
    creator,
    did,
    from,
  }) {
    const metadata = {
      name: name,
      description: description,
      image: image,
      sources: sources,
      owner: owner,
      parent: parent,
      verified_credential_ref: verifiedCredentialRef,
      links: links,
      creator: creator,
      did: did,
      from: from,
    }

    const sub = new Subject<string>()
    const query = `message.action='Metadata'`

    const subscription = this.anconClient.tm.subscribeTx(query).subscribe({
      next: async (log: any) => {
        try {
          // Decode response
          const res = MsgMetadataResponse.decode(log.result.data)

          // Hack: Protobuf issue
          const cid = res.cid.substring(14)
          sub.next(cid)
          sub.complete()
        } catch (err) {
          sub.error(err)
        }
      },
    }) as Subscription

    try {
      const wait = new Promise((resolve, reject) => {
        sub.subscribe({ next: (i) => resolve(i), error: (e) => reject(e) })
      })
      const encoded = this.anconClient.msgService.msgMetadata(
        metadata,
      )
      // hack
      encoded.value.creator = creator;
      encoded.value.sources = JSON.stringify(encoded.value.sources)
      encoded.value.links = JSON.stringify(encoded.value.links)
      const tx = await this.anconClient.msgService.signAndBroadcast([encoded], {
        fee: {
          amount: [
            {
              denom: 'token',
              amount: '1',
            },
          ],
          gas: '5000000',
        },
      })
      const cid = await wait
      subscription.unsubscribe()
      return { transaction: tx, cid }
    } catch (e) {
      throw e
    }
  }
  /**
   * Add Ancon File
   * @param
   */
  async addFile(did: string, payload: File) {
    let result = await this.blobToKeccak256(payload)

    const file = {
      path: payload.name,
      did: did,
      from: this.account,
      time: payload.lastModified.toString(),
      mode: '',
      content: base64.encode(result.content),
      contentType: payload.type,
      creator: this.account,
    }
    const sub = new Subject<string>()
    const query = `message.action='File'`

    const subscription = this.anconClient.tm.subscribeTx(query).subscribe({
      next: async (log: any) => {
        try {
          // Decode response
          const res = MsgFileResponse.decode(log.result.data)

          // Hack: Protobuf issue
          const cid = res.hash.substring(10)
          sub.next(cid)
          sub.complete()
        } catch (err) {
          sub.error(err)
        }
      },
    }) as Subscription

    try {
      const wait = new Promise((resolve, reject) => {
        sub.subscribe({ next: (i) => resolve(i), error: (e) => reject(e) })
      })
      const encoded = this.anconClient.msgService.msgFile(file)
      const tx = await this.anconClient.msgService.signAndBroadcast([encoded], {
        fee: {
          amount: [
            {
              denom: 'token',
              amount: '1',
            },
          ],
          gas: '5000000',
        },
      })
      const cid = await wait
      subscription.unsubscribe()
      return { transaction: tx, cid }
    } catch (e) {
      throw e
    }
  }

  /**
   * Get IPLD object
   * @param cid content id
   */
  async getObject(cid: string): Promise<any> {
    let temp = await this.anconAPI.file.get(cid)

    return temp
  }
}
