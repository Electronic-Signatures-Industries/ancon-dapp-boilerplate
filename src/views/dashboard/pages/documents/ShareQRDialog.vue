<template>
  <v-dialog
    persistent
    max-width="700px"
    :value="show"
    @update:value="updateShow"
  >
    <v-card>
      <v-card-title>Compartir Documento</v-card-title>

      <v-card-text>
        <v-row>
          <v-col cols="4">
            <qrcode
              v-if="this.documentMetadata && this.valueString"
              v-model="this.valueString"
            />
          </v-col>

          <v-col cols="8">
            <v-list>
              <v-list-item two-line>
                <v-list-item-content>
                  <v-list-item-title>DID</v-list-item-title>
                  <v-list-item-subtitle>{{ this.didID }}</v-list-item-subtitle>
                </v-list-item-content>
              </v-list-item>

              <v-list-item two-line>
                <v-list-item-content>
                  <v-list-item-title>IPLD Content ID</v-list-item-title>
                  <v-list-item-subtitle>{{ this.contentRef }}</v-list-item-subtitle>
                </v-list-item-content>
              </v-list-item>
            </v-list>
          </v-col>
        </v-row>
      </v-card-text>

      <v-card-actions>
        <v-spacer />

        <v-btn
          text
          @click="() => updateShow(false)"
        >
          Cerrar
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<style lang="scss" scoped>
.v-list-item__subtitle {
  white-space: normal;
}
</style>

<script lang="ts">
import Vue from 'vue'
import Component from 'vue-class-component';
import { Emit, Prop } from 'vue-property-decorator';
import VueQrcode from '@chenfengyuan/vue-qrcode';
import { DocumentMetadata } from '../wallet/IPFSManager';
import { DID } from 'dids';

@Component({
  components: {
    [VueQrcode.name]: VueQrcode
  }
})
export default class ShareQRDialog extends Vue {
  @Prop()
  readonly show: boolean;

  @Prop()
  readonly documentMetadata?: DocumentMetadata;

  @Prop()
  readonly did?: DID;

  @Emit("update:show")
  updateShow(newValue: boolean) {
    return newValue;
  }

  get contentRef(): string {
    return this.documentMetadata?.contentRef ?? '';
  }

  get valueString(): string | null {
    if (!this.did) return null;
    return `ipld:${this.contentRef}|${this.didID}`;
  }

  get didID(): string {
    return this.did?.id ?? '';
  }
}
</script>