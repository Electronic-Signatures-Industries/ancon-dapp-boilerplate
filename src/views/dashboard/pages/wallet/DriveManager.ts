import { DID } from 'dids';
import { forkJoin } from 'rxjs';
import { DIDManager } from './DIDManager';
import { IPFSManager } from './IPFSManager';

export class DriveManager {
    constructor(
        private ipfs: IPFSManager,
        private did: DID        
    ) {

    }

    async appendDocumentSet(files: File[]) {
        // get parent node
        const parent = await this.ipfs.getCurrentNode();

        // get promises of signed objects
        const pushFiles = files.map(async (f) => {
            return await this.ipfs.addSignedObject(this.did, f);
        });

        // run promises
        const items = await forkJoin(pushFiles).toPromise();

        // get indexed of ipfs
        const parentindex = await this.ipfs.addIndex(this.did, items, parent);

        // response with indexes (IPFS cid's)
        const res = await this.ipfs.setCurrentNode(
            parentindex,
        );
        debugger;
        return res.name;
    }
}