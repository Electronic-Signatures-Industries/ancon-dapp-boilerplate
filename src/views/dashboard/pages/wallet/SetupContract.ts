import { ethers } from 'ethers'
import { DIDManager } from './DIDManager'
import { DriveManager } from './DriveManager'
import { IPFSManager } from './IPFSManager'

// @ts-ignore
const mainContracts = require('./abi/main')
const minterAbi = require('./abi/NFTDocumentMinter')

export const bindContracts = async (
  web3: any,
  address: string,
  didManager: DIDManager,
  ipfs: IPFSManager,
) => {
  const provider = new ethers.providers.Web3Provider(web3)
  const did = await didManager.create3ID(address)
  const driveManager = new DriveManager(ipfs, did)

  localStorage.setItem('did:' + address, did.id)
  const NFTFactory = new ethers.Contract(
    mainContracts.NFTFactory.address.bsctestnet,
    mainContracts.NFTFactory.raw.abi,
    provider.getSigner(),
  )
  const DocumentAnchoring = new ethers.Contract(
    mainContracts.DocumentAnchoring.address.bsctestnet,
    mainContracts.DocumentAnchoring.raw.abi,
    provider.getSigner(),
  )
  const DAI = new ethers.Contract(
    mainContracts.TestDAI.address.bsctestnet,
    mainContracts.TestDAI.raw.abi,
    provider.getSigner(),
  )
  const getNFTDocumentMinter = (address) =>
    new ethers.Contract(address, minterAbi, provider.getSigner())
  return {
    driveManager,
    did,
    contracts: {
      NFTFactory,
      DocumentAnchoring,
      DAI,
      getNFTDocumentMinter,
    },
  }
}
