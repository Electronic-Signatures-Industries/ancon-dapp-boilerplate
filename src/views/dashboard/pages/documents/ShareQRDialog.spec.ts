import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import { DocumentMetadata } from "@/views/dashboard/pages/wallet/IPFSManager";
import ShareQRDialog from "./ShareQRDialog.vue";

const documentMetadata: DocumentMetadata = {
  name: "Mocked Document Name",
  contentType: "application/json",
  timestamp: 1000,
  contentRef: "Mock-Content-Ref",
};

const mockDID = {
  id: "did:mock:1",
};

describe("ShareQRDialog.vue", () => {
  let component: Wrapper<ShareQRDialog>;

  beforeEach(async () => {
    const localVue = createLocalVue();

    component = shallowMount(ShareQRDialog, {
      localVue,
      vuetify: new Vuetify(),
    });
  });

  describe("Empty Component", () => {
    it("should match snapshot", async () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe("Component with data", () => {
    beforeEach(async () => {
      component.setProps({
        show: true,
        documentMetadata,
        did: mockDID,
      });

      await component.vm.$nextTick();
    });

    it("should match snapshot when data is provided", async () => {
      expect(component).toMatchSnapshot();
    });
  });
});
