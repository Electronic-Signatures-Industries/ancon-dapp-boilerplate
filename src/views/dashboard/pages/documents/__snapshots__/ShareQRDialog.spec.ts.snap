// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareQRDialog.vue Component with data should match snapshot when data is provided 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="700px" origin="center center" persistent="true" retainfocus="true" transition="dialog-transition" width="auto" value="true">
  <v-card-stub loaderheight="4" tag="div">
    <v-card-title-stub>Compartir Documento</v-card-title-stub>
    <v-card-text-stub>
      <v-row-stub tag="div">
        <v-col-stub cols="4" tag="div">
          <qrcode-stub value="ipld:Mock-Content-Ref|did:mock:1" tag="canvas"></qrcode-stub>
        </v-col-stub>
        <v-col-stub cols="8" tag="div">
          <v-list-stub tag="div">
            <v-list-item-stub activeclass="" tag="div" twoline="true">
              <v-list-item-content-stub>
                <v-list-item-title-stub>DID</v-list-item-title-stub>
                <v-list-item-subtitle-stub>did:mock:1</v-list-item-subtitle-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" twoline="true">
              <v-list-item-content-stub>
                <v-list-item-title-stub>IPLD Content ID</v-list-item-title-stub>
                <v-list-item-subtitle-stub>Mock-Content-Ref</v-list-item-subtitle-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-stub>
        </v-col-stub>
      </v-row-stub>
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub tag="button" activeclass="" text="true" type="button">
        Cerrar
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;

exports[`ShareQRDialog.vue Empty Component should match snapshot 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="700px" origin="center center" persistent="true" retainfocus="true" transition="dialog-transition" width="auto">
  <v-card-stub loaderheight="4" tag="div">
    <v-card-title-stub>Compartir Documento</v-card-title-stub>
    <v-card-text-stub>
      <v-row-stub tag="div">
        <v-col-stub cols="4" tag="div">
          <!---->
        </v-col-stub>
        <v-col-stub cols="8" tag="div">
          <v-list-stub tag="div">
            <v-list-item-stub activeclass="" tag="div" twoline="true">
              <v-list-item-content-stub>
                <v-list-item-title-stub>DID</v-list-item-title-stub>
                <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
            <v-list-item-stub activeclass="" tag="div" twoline="true">
              <v-list-item-content-stub>
                <v-list-item-title-stub>IPLD Content ID</v-list-item-title-stub>
                <v-list-item-subtitle-stub></v-list-item-subtitle-stub>
              </v-list-item-content-stub>
            </v-list-item-stub>
          </v-list-stub>
        </v-col-stub>
      </v-row-stub>
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub tag="button" activeclass="" text="true" type="button">
        Cerrar
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
